activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -359,6 +359,22 @@ def transaction(requires_new: nil, isolation: nil, joinable: true, &block)
           end
           yield current_transaction.user_transaction
         else
+          # If we're running inside the single, non-joinable transaction that
+          # ActiveRecord::TestFixtures starts around each example (depth == 1),
+          # an `isolation:` hint must be validated then ignored so that the
+          # adapter isn't asked to change the isolation level mid-transaction.
+          if isolation && open_transactions == 1 && !current_transaction.joinable?
```

Comment:
Maybe it's time we add some more explicit tagging on the test fixtures transaction? I never felt like checking for `joinable? == false` was very clear.

@matthewd what would you think about having a (private API) method such as `current_transaction.fixtures?` (I can't come up with a better name, but there must be one).

Reply:
Or perhaps, we could define a `Transaction` subclass? And check the type?

Reply:
Yeah... `joinable` isn't public API, so technically (despite feeling gross every time I see it) it is valid for us to treat `!joinable?` == `outer_fixture_transaction_we_should_sometimes_pretend_isnt_there?`... but it seems nice to have a clearer direct representation of what we're asking. (`joinable?` was an elegant abstraction originally, when the fixture side was imposing the logic, and just wanted to request one specific behaviour from the transaction... but as soon as started using that in reverse to infer the transaction's purpose, it became overloaded.)